pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -269,18 +267,13 @@
     private Channel listenChannelTls;
 
     private boolean preciseTopicPublishRateLimitingEnable;
-    private final long maxMessagePublishBufferBytes;
-    private final long resumeProducerReadMessagePublishBufferBytes;
-    private volatile boolean reachMessagePublishBufferThreshold;
+    private final AtomicInteger pausedConnections = new AtomicInteger();
```

Comment:
The reading frequency here should be very low, only used in unit tests or metrics, why not use longAdder

Reply:
Good point, changed that.